specs/core/0_beacon-chain.md

```diff
@@ -1065,70 +1147,73 @@ def change_validators(validators: List[ValidatorRecord], current_slot: int) -> N
     withdrawable_validators = sorted(filter(withdrawable, validators), key=lambda v: v.exit_seq)
     for v in withdrawable_validators[:WITHDRAWALS_PER_CYCLE]:
         if v.status == PENALIZED:
-            v.balance -= v.balance * min(total_penalties * 3, total_balance) // total_balance
+            v.balance -= balance_at_stake(v) * min(total_penalties * 3, total_balance) // total_balance
         v.status = WITHDRAWN
         v.last_status_change_slot = current_slot
 
         withdraw_amount = v.balance
-        ...
-        # STUB: withdraw to shard chain
+        # STUB: withdraw to shard chain    
 ```
 
-* Set `state.validator_set_change_slot = state.last_state_recalculation_slot`
-* Set `shard_and_committee_for_slots[:CYCLE_LENGTH] = shard_and_committee_for_slots[CYCLE_LENGTH:]`
-* Let `next_start_shard = (shard_and_committee_for_slots[-1][-1].shard + 1) % SHARD_COUNT`
-* Set `shard_and_committee_for_slots[CYCLE_LENGTH:] = get_new_shuffling(state.next_shuffling_seed, validators, next_start_shard)`
+And perform the following updates to the `state`:
+
+* Set `state.validator_set_change_slot = s`
```

Comment:
This actually needs to be `state.validator_set_change_slot = s + CYCLE_LENGTH`

Because we want to capture that it happened _now_ rather than a cycle ago, and we have't yet updated the `last_state_recalculation_slot`